theano/gpuarray/multinomial.py

```diff
@@ -327,6 +327,7 @@ def gpu_kernels(self, node, name):
 
                     if (! %(replace)s )
                     {
+                        __syncthreads();
```

Comment:
I'm not completely sure.
Each thread is supposed to sample something at each iteration, otherwise it means we have un-initialized memory in `global_outs`.
However, if there are completely inactive threads due to scheduling, I'm not sure.